src/container/srv_epoch.c

```diff
@@ -162,36 +228,66 @@ snap_create_bcast(struct rdb_tx *tx, struct cont *cont, uuid_t coh_uuid,
 		rc = -DER_IO;
 		goto out_rpc;
 	}
-
 	*epoch = in->tsi_epoch;
+	/* oit oids index kvs not existed for containers created before release2.4 */
+	if (cont_ver >= 2) {
+		d_iov_set(&key, epoch, sizeof(*epoch));
+		oov.oov_id = in->tsi_oit_oid;
+		oov.oov_ver = pool_map_get_version(cont->c_svc->cs_pool->sp_map);
```

Comment:
Hmm, I thought we need store the layout version, or oov_id needs to be daos_unit_oid_t, Sorry, I forgot why do we need pool map version here? or layout version?

Do we need provide an oit_table delete api? since these oit table are only temporary for me?  

Btw: will snapshot delete will delete related oit table?

Reply:
Currently, snapshot destroy will delete OIT table kv record.

Reply:
> Hmm, I thought we need store the layout version, or oov_id needs to be daos_unit_oid_t, Sorry, I forgot why do we need pool map version here? or layout version?
> 
> Do we need provide an oit_table delete api? since these oit table are only temporary for me?
> 
> Btw: will snapshot delete will delete related oit table?

I removed pool map version now, i don't see use case to store extra layout version, daos_oit_xxx api expect value is daos_obj_id_t, we need allocate extra memory and filter to only return daos_obj_id  for this, I'll try to push async mode support and oit_table delete api together in a separated PR.